src/build.rs

```diff
@@ -123,20 +123,21 @@ impl BuildSystem {
             let entry = entry.context("error reading contents of staging dist dir")?;
             let target_path = final_dist.join(entry.file_name());
 
-            fs::rename(entry.path(), target_path)
+            fs::rename(entry.path(), &target_path)
                 .await
-                .context("error moving from staging dir to dist dir")?;
+                .with_context(|| format!("error moving {:?} to {:?}", &entry.path(), &target_path))?;
         }
         Ok(())
     }
 
+    /// Clean the contents of the final dist dir.
     async fn clean_final(&self) -> Result<()> {
         let final_dist = self.cfg.final_dist.clone();
 
         let mut entries = fs::read_dir(&final_dist).await.context("error reading final dist dir")?;
         while let Some(entry) = entries.next().await {
             let entry = entry.context("error reading contents of final dist dir")?;
-            if entry.file_name() == ".stage" {
+            if entry.file_name() == STAGE_DIR {
```

Comment:
Should we canonicalize the `file_name`, perhaps? I fear another regression like #84 from not canonicalizing the path

Reply:
I believe file name is just the final part when splitting by directory separator; it shouldn’t be an actual path in any circumstances, just a stem and extension.

Reply:
Yea, agreed with Philip's analysis here. I think we are g2g in this case.